tests/phpunit/tests/rest-api/wpRestTemplatesController.php

```diff
@@ -678,4 +684,91 @@ protected function find_and_normalize_template_by_id( $templates, $id ) {
 		return null;
 	}
 
+	/**
+	 * @ticket 56467
+	 * @covers WP_REST_Templates_Controller::create_item
+	 */
+	public function test_create_item_with_is_wp_suggestion() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'POST', '/wp/v2/templates' );
+		// `is_wp_suggestion` true.
+		$body_params = array(
+			'slug'             => 'page-rigas',
+			'description'      => 'Just a description',
+			'title'            => 'My Template',
+			'content'          => 'Content',
+			'is_wp_suggestion' => true,
+			'author'           => self::$admin_id,
+		);
+		$request->set_body_params( $body_params );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		unset( $data['_links'] );
+		unset( $data['wp_id'] );
+		$expected = array(
+			'id'             => 'default//page-rigas',
+			'theme'          => 'default',
+			'content'        => array(
+				'raw' => 'Content',
+			),
+			'slug'           => 'page-rigas',
+			'source'         => 'custom',
+			'origin'         => null,
+			'type'           => 'wp_template',
+			'description'    => 'Just a description',
+			'title'          => array(
+				'raw'      => 'My Template',
+				'rendered' => 'My Template',
+			),
+			'status'         => 'publish',
+			'has_theme_file' => false,
+			'is_custom'      => false,
+			'author'         => self::$admin_id,
+		);
+		$this->assertSame( $expected, $data );
+		// `is_wp_suggestion` false.
+		$body_params = wp_parse_args(
+			array(
+				'slug'             => 'page-hi',
+				'is_wp_suggestion' => false,
+			),
+			$body_params
+		);
+		$request->set_body_params( $body_params );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		unset( $data['_links'] );
+		unset( $data['wp_id'] );
+		$expected = wp_parse_args(
+			array(
+				'id'        => 'default//page-hi',
+				'slug'      => 'page-hi',
+				'is_custom' => true,
+			),
+			$expected
+		);
+		$this->assertSame( $expected, $data );
+	}
+
+	/**
+	 * @ticket 56467
+	 * @covers WP_REST_Templates_Controller::get_template_fallback
+	 */
+	public function test_get_template_fallback() {
+		wp_set_current_user( self::$admin_id );
+		switch_theme( 'block-theme' );
+		$request = new WP_REST_Request( 'GET', '/wp/v2/templates/lookup' );
+		// Should fallback to `index.html` .
+		$request->set_param( 'slug', 'tag-status' );
+		$request->set_param( 'is_custom', false );
+		$request->set_param( 'template_prefix', 'tag' );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEquals( 'index', $response->get_data()->slug );
+		// Should fallback to `page.html` .
+		$request->set_param( 'slug', 'page-hello' );
+		$request->set_param( 'is_custom', false );
+		$request->set_param( 'template_prefix', 'page' );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEquals( 'page', $response->get_data()->slug );
```

Comment:
* use `assertSame()` instead of `assertEquals()`
* add a message to each assertion